### ISSUE TITLE
Copy-ServiceFabricApplicationPackage : Operation timed out.

### ISSUE THREAD

**Issue by sondreb** [Azure/service-fabric-issues#10]
It is not possible to deploy to cluster running on Azure that are of a certain size. Keep getting "Operation timed out". From what I have found while searching for a solution, is that there is a 10 minute limit for the package deploy.

Have tried to create a VM in the same data center as the cluster, but even this is not able to deploy our application within the timeout threshold.

When can we expect an increase of this timeout threshold?

The only solution I have found so far, is to create an intricate and complex process of first deploying all services with an dummy .exe, then have scripts that loop through all individual services, populates the Code folder with the real files, upgrades version numbers on ApplicationManifest.xml and ServiceManifest.xml. Then deploy the package again, provision a new application type with the new version, then perform an upgrade. This has to go on for each individual service package, which are around 10MB to 80MB each. The whole process takes an incredible amount of time, and it's a messy and complex code to do all of this.

**Comment by cwe1ss** [comment]
We also have this issue from time to time. Our packages usually are about 30-40 MB. We got this when we deployed from our local dev machine, but we also got this error when we deployed from VSTS (so within Azure). For this reason, I don't believe it's due to slow upload times. 

Maybe there could be something wrong within SF that blocks the upload? (like a pending replication action, ...)

**Comment by sondreb** [comment]
There is nothing wrong with SF by itself, I'm able to deploy the app if I reduce it to a single small service. So no upgrade/pending operations that is blocking it. Tried on two new clusters, both fails the same way.

**Comment by cwe1ss** [comment]
yep - could be two completely unrelated issues.

**Comment by seanmck** [comment]
We will be removing the hard-coded 10 minute timeout in an upcoming release - likely in July.

**Comment by aelij** [comment]
As a workaround for the time being you can edit the files `Publish-NewServiceFabricApplication.ps1` and `Publish-UpgradedServiceFabricApplication.ps1` in `C:\Program Files\Microsoft SDKs\Service Fabric\Tools\PSModule\ServiceFabricSDK` and add `-TimeoutSec 600` to the `Register-ServiceFabricApplication` command (600 is apparently the max allowed). Note that you may need to apply these changes after each SDK update.

**Comment by sondreb** [comment]
@aelij Thanks for the comment, but that does not help as the error is with the copying of the image (service packages).

"Copy-ServiceFabricApplicationPackage" is called before Register-ServiceFabricApplicationType, and I don't think the Copy operation takes a timeout parameter?

At least not the .NET library (Service.Fabric.dll) have overloads with timeout support for the package copying.

**Comment by aelij** [comment]
@sondreb You're right, I missed the name of the command. We kept getting timeouts on "register".

The only other thing I can think of is for you to perform the upload from either a VM in the same data center (we've set up a build server that way which does automatic deployments) or even directly on one of the cluster machines.

**Comment by LightCrystal** [comment]
Any timeframe on when this will be available? I have a ready to production application but it's more than 2GB with some local DB, and cannot put it live cause of this. Any real workaround in the meantime?

**Comment by shlatchz** [comment]
@LightCrystal Try deploying it from an Azure VM on the same DataCenter

**Comment by seanmck** [comment]
We are hoping to get it out in the next week.

**Comment by ruijarimba** [comment]
Hi @seanmck , any updates on this issue?

**Comment by mkosieradzki** [comment]
Issue still exists in 5.2.207. (~200 MB package)

**Comment by mkosieradzki** [comment]
This bug is much worse in 5.2.207. Copy-ServiceFabricApplication has received desired parameter -TimeoutSec however it is not used in Publish-NewServiceFabricApplication.ps1.
The new version of Copy-ServiceFabricApplication has also new default value for this parameter which is very low ~1-2 minutes.
To workaround this limitation (WARNING: EVIL WORKAROUND): edit C:\Program Files\Microsoft SDKs\Service Fabric\Tools\PSModule\ServiceFabricSDK\Publish-NewServiceFabricApplication.ps1.
Replace line 230:
Copy-ServiceFabricApplicationPackage -ApplicationPackagePath $AppPkgPathToUse -ImageStoreConnectionString $imageStoreConnectionString -ApplicationPackagePathInImageStore $applicationPackagePathInImageStore
with
Copy-ServiceFabricApplicationPackage -ApplicationPackagePath $AppPkgPathToUse -ImageStoreConnectionString $imageStoreConnectionString -ApplicationPackagePathInImageStore $applicationPackagePathInImageStore -TimeoutSec 600

Feel free to increase 600 to whatever you need.

WARNING: You are altering SDK files, You break digital signature of this file.

**Comment by tillmaurer** [comment]
Having the same problem in version 5.2.207.9590. Trying to upload a 100 MB package fails within 1-2 min.
Copy-ServiceFabricApplicationPackage : Operation timed out.

**Comment by masnider** [comment]
@tillmaurer did you modify the timeout? Modifying the timeout is the recommended mechanism and is not quite as evil as is suggested by @mkosieradzki :) Folks needing to modify the deployment scripts for their scenario is a big part of why they're included as a part of the solution and not just baked into VS or the product.

**Comment by marcuslind90** [comment]
@masnider When you say "Modifying the timeout" - How do you do that without modyfing the file that @tillmaurer is suggesting? Is there some config file somewhere that I've missed?

**Comment by masnider** [comment]
@marcuslind90 no, _modify the file_. As I said:

> Folks needing to modify the deployment scripts for their scenario is a big part of why they're included as a part of the solution and not just baked into VS or the product.

The Publish script is provided as an example of what you would need to do in order to set up a "regular" deployment. It's included so you can see all the steps and if necessary modify them. The default script which is provided does an ok job, but there are lots of cases where you may need or want to modify the steps it takes or the exact parameters it passes to the actual APIs. For example, if you have a rather large application package (or just a slower internet connection), or if you want to target a different image store location, multiple environments, kick off some CI/CD work, etc, those are some examples where you'd probably modify it.

**Comment by marcuslind90** [comment]
I was able to fix our issue with the Operation Timeout by publishing with Visual Studio 2015, and checking the 'Upgrade the Application' in the Publish window. When checking the box it seems as if it uses the 'Upgrade Settings' which you can configure there and set the Timeout to 'never'. 

When publishing for first time, and not choosing 'Upgrade Application' it seems as if you can not set the configuration of the Timeout inside Visual Studio 2015.

**Comment by masnider** [comment]
The timeout there is about the total time the upgrade operation takes, not about the time it takes to copy or register the application package itself. Again, modifying the Publish script to specifically pass in the timeouts you require is the recommended way forward.

**Comment by bklooste** [comment]
Have set both to 600 and still get issues not so much the copy (from a fast pipe or vm)  but the register.. This may relate these are .net core angular  apps and event a trivial single page will have a thousand files.   

http://stackoverflow.com/questions/38119341/register-servicefabricapplicationtype-on-a-secure-cluster-always-times-out

**Comment by bklooste** [comment]
Changing the cluster as per the above link did work for the register is there a way to automate that for cluster deploy .. bit annoying that its not documented anywhere  (i was using a VM  ) .

**Comment by cosminpopescu** [comment]
Updated the Publish-NewServiceFabricApplication.ps1 worked for me. This should be a supported parameter.

**Comment by masnider** [comment]
Supported parameter where?

**Comment by MikkelHegn** [comment]
In the new 2.3 release of the SDK, a new parameter CopyPackageTimeoutSec is added to the Deploy-FabricApplication.ps1 script, that is part of the Visual Studio Service Fabric project.
The same is added on SF SDK. If you want to use this parameter from within Visual Studio, you need to update the Deploy-FabricApplication.ps1 with the required timeout value.

The script does not set any timeout per default, meaning it will not override the 600 sec timeout specified in the Publish-UpgradedServiceFarbicApplication.ps1 script from the SDK.

To complete the story, the timeout is what get's passed along to the Copy-ServiceFabricApplicationPackage CmdLet as the -TimeOutSec parameter.

**Comment by lanfeust69** [comment]
@MikkelHegn : That's definitely a step in the right direction, but as mentioned by @bklooste, it would be nice to have the same thing for the TimeoutSec parameter of the Register-ServiceFabricApplicationType cmdlet.

**Comment by cwe1ss** [comment]
Are there plans to compress (zip) the packages? It seems like a waste of time and resources to copy everything unzipped. Ideally, SF itself would also store the packages compressed to increase the speed when a node has to grab the package from the image store.

**Comment by MikkelHegn** [comment]
@lanfeust69 Are you asking for having that parameter exposed in the Deploy-FabricApplication.ps1 script as well? Are you currently hitting the default timeout when registering application types?

For now, we have no plans, since most issues we are getting reported are in terms of copying packages to the cluster.

The Register-ServiceFabricApplicationType cmdlet already exposes a TimeOutSec parameter for you to use.

**Comment by lanfeust69** [comment]
@MikkelHegn Yes, we sometimes reach the 2' timeout on our on-prem dev cluster. It's composed of three rather small VMs, so that things can be a bit slow if there is some load. When it succeeds, it takes at least 1'30, not very much to spare. Admittedly, we are still very much on a "lift and shift" approach, with our services rather small in scope, but using most of the existing assemblies, so that the resulting package is much bigger than it should be.
Even if not critical at all, it seems only logical to be consistent and expose both timeouts, at least on the `Publish-Xxx` scripts, so that we can tune `Deploy-FabricApplication.ps1`, which is in the application source, without changing the SDK (which is not), or duplicating it.

**Comment by philcarbone** [comment]
For those of you running into this issue on VSTS/VSO/TFS, I've posted a request to update the deployment task here: https://github.com/Microsoft/vsts-tasks/issues/2894

**Comment by laultman** [comment]
I just up graded to Service Fabric SDK version 2.3.301.9590 and Service Fabric runtime version 5.3.301.9590. When I loaded my solution VS 2015 Update 3 updated my service fabric project to the current version.  The publish still fails with the time out error.  It also fails when I try to use PowerShell to copy the package to the image store with the same error.  I am attempting this from different locations with upload speeds of >6mbs.
This is a long standing issue that needs to be fixed.

**Comment by MikkelHegn** [comment]
What timeout value did you provide to the deployment script? If you did not provide any, the default is still 600 sec. You can set it in the Deploy-FabricApplication.ps1 script in your project.

**Comment by moncapitaine** [comment]
Copy seems not to be the problem in my case. Registration takes far too long. Why it takes >2 minutes for registration? I am not interested in changing the timeouts but am curious about the reason for >2 minutes for registration?

**Comment by masnider** [comment]
Registration causes Service Fabric to do several things
- We digest the application package to process it. This is where service fabric learns about the different service types, the packages, their versions, etc. We also look at files and create checksums and perform other validations. 
- In real clusters, we replicate the application package to several nodes. 

The larger and more complicated the package the longer both of these steps will take.

**Comment by edtheshed** [comment]
I am getting this issue when trying to deploy our service fabric application, but it is when I am using the task in release management in VSTS. It seems to timeout after two minutes. How can I increase the timeout amount in VSTS?

Also, my application package is just over 2 GB in size...

**Comment by masnider** [comment]
@edtheshed did the suggested workarounds here and in the linked issue not work for you?

**Comment by edtheshed** [comment]
Well as it is in VSTS, I don't actually have access to the script that is being used to deploy the application. Unless I have missed a comment saying there is a way to fix this method of deploying?

**Comment by pratapbhaskar** [comment]
Well its getting really hard to deploy Service Fabric through a CD/CI pipeline. We have a package which is ~200MB, the CD pipeline was meant to upgrade the application, ALAS it hasn't been able to move forward from Copy-ServiceFabricApplicationPackage, it always times out. We have tried moving the buildserver to the same data center as Cluster, update the timeout sec parameter in the scripts. 

Please do something about these time outs.

**Comment by MikkelHegn** [comment]
There is a fix coming to expose the copy and register timeout in the VSTS Service Fabric Deployment task. The PR for the fix is here: https://github.com/Microsoft/vsts-tasks/pull/3137. The fix will be rolled-out with VSTS quarterly updates, which makes it to be available around end of year.

If you need the fix sooner, you can clone the repo with the fix, build your own extension and deploy it to your VSTS instance. Use these guidelines: https://github.com/Microsoft/vsts-tasks/blob/master/docs/contribute.md and here https://github.com/Microsoft/tfs-cli/blob/master/docs/extensions.md

Remember if you break out and use your own task, you do not get our updates, but the official task will still be available.

If you see other improvements - please consider contributing to the repo :-)

**Comment by MikkelHegn** [comment]
@pratapbhaskar What timeout value are you using when this fails?

**Comment by EricJizbaMSFT** [comment]
Just to clarify a few points made by Mikkel about building and uploading the VSTS task to your own instance (if you can't wait for it to get to you automatically):

1. You do not need to package the task in an extension. You can upload just the task itself. Follow these directions to upload: https://github.com/Microsoft/tfs-cli/blob/master/docs/buildtasks.md
2. You *will* get future updates to the task. We will overwrite whatever version you uploaded
3. Hosted agents are still being updated with the 2.3 version of the Service Fabric SDK, so you will either have to use a custom agent or wait a few more weeks

**Comment by cwe1ss** [comment]
I think it's important to also focus on the core issue. Increasing the timeout is a nice quick-fix but it doesn't solve the problem. Our packages usually only have 30-40 MB and we still often get timeouts. 

This is super frustrating, especially when you need to do an important deployment! My colleagues are constantly complaining about this which makes Service Fabric a hard sell sometimes :(

Again, it would be great if you could use compressed files for everything - for deployments and when you copy stuff onto a machine in the cluster. With our application packages, this would reduce the size to ~ 1/3.

**Comment by pratapbhaskar** [comment]
@MikkelHegn I had modified the Publish-UpgradedServiceFabricApplicatio module. For Copy-ServiceFabricApplication command the timeout was 900 and for Register-Application it was 600 (anything above 600, i get an exception). We use Jenkins for out CD pipeline.

**Comment by vaishnavk** [comment]
@pratapbhaskar Is Copy-ServiceFabricApplicationPackage failing very often? Have you checked the system service health to ensure they are healthy?

**Comment by pratapbhaskar** [comment]
@vaishnavk yes Copy command fails often, If lets say Copy passes then Register-ServiceFabricApplication fails.

**Comment by vaishnavk** [comment]
We will need access to fabric logs to see the why copy is failing for your cluster. Is there a way you can share your account name and a sas key for the storage account holding fabric logs?

**Comment by moncapitaine** [comment]
For me: It's the local cluster "c:/SfDevCluster". Which log files you want? I can post it if you want.

**Comment by pratapbhaskar** [comment]
@vaishnavk is there any secure way i could share the logs. And one more update after changing the copy timeout to 900 and register time out to 600, CD jobs are passing more than often. Now i have to deal with Jenkins slave time out..Ugh!!!

**Comment by MikkelHegn** [comment]
To give a more general feedback on this, these are the common issues we see in support:
-	Copy failures on large packages (slow link speeds)
-	Copy and Provision failure on relatively small packages with large number of files (ESE cursor issues mentioned in the stackoverflow http://stackoverflow.com/questions/38119341/register-servicefabricapplicationtype-on-a-secure-cluster-always-times-out)
-	Provision failures in the large packages (the timeout must be increased, but often time this is the result of the large number of files)

Here are the improvements we are working on:
- Updating the CmdLets to give you progress on Copy and Provision (Estimated release - 2/2017)
- Ability to copy compressed packages (Estimated release - 2/2017)
- Remove hard coded timeouts for copy and register (Estimated release - 12/2016)

I'm going to keep this issue open till we have the above released.

**Comment by cwe1ss** [comment]
Thank you for the progress update!! Really looking forward to compressed packages! 👍

**Comment by dmaltsiniotis** [comment]
Hi there, any updates on this?
  Remove hard coded timeouts for copy and register (Estimated release - 12/2016)

This has become a production-level blocker for us, and if possible I need access to a pre-release version of the cluster or some hack/workaround that removes these limits. I know fiddling with the SF manifest is not supported or recommened, but if I can somehow override the 10 min limit, well then that's what I need to do. Our project has grown requiring a >10 minute registration process, but alas the cluster won't allow it:

Start-ServiceFabricApplicationUpgrade : Client timeout must be less than or equal to the cluster configuration for MaxOperationTimeout. Requested value and timeout limit: (29:59.999, 10:00.000)

Cheers,

-Demetri

**Comment by alexwun** [comment]
Did you mean to increase the timeout on Register-ServcieFabricApplicationType rather than Start-ServiceFabricApplicationUpgrade?

**Comment by dmaltsiniotis** [comment]
@alexwun Yes my bad :) I was in an rush to show the output example from Visual Studio and copied the wrong output, however I _think_ I need Start updated as well. We have around 60 actors grouped in various app, and the largest app just won't deploy anymore after adding a few more actors to it. I've tried adjusting every timeout I could find so far.

**Comment by alexwun** [comment]
Copy-ServiceFabricApplicationPackage and Register-ServiceFabricApplicationType are the two commands that will need higher timeouts if the application package has gotten large. Start-ServiceFabricApplicationUpgrade does not need a high timeout value for the command itself since all it does is send a message into the system for starting the upgrade.

Does increasing the timeouts for Copy and Register, but leaving the default timeout for Start-Upgrade unblock you?

**Comment by dmaltsiniotis** [comment]
tldr; Unfortunately no.

Register-ServiceFabricApplicationType is failing because the requested timeout value of 1800 seconds is greater than the MaxOperationTimeout value. I need registering the app to be allowed to take more than 10 minutes.

More details:

I have set the timeout for CopyPackageTimeoutSec in the default Deploy-FabricApplication.ps1 file to the maximum allowable 1800 seconds. This parameter seems to pass through to Publish-UpgradedSericeFabricApplications.ps1, and sets the TimeOutSec value for Copy-ServiceFabricApplicationPackage. And indeed, our copy operations always succeed (the size of the packages isn't that large), it's just registering that seems to be the problem.

Register-ServiceFabricApplicationType (by default script in the Runtime/SDK v5.4.145/v2.4.145) only passes through -ApplicationPathInImageStore and not pass-through any timeout settings. It looks like the change (https://github.com/Microsoft/vsts-tasks/pull/3137) was merged into the master branch to add the Register timeout as part of the SDK publish example scripts, but it did not make it to the 2.4.145 SDK release?

I did as @MikkelHegn suggested, and used these SDK scripts as a basis for creating a totally custom deployment process/scripts. However, I cannot request anything greater than 600 seconds for Register-ServiceFabricApplicationType (see above error)

I am forced to split out the actors into two different apps to alleviate this issue for now, I will see if that works, but I have to refactor a lot of code.

**Comment by MikkelHegn** [comment]
Are you in a VSTS scenario or another place? The CmdLet Register-ServiceFabricApplicationType supports the TimeOutSec parameter in 5.4, so if you're not using VSTS you should be able to squeeze it in there, by changing the SDK ps functions and the Deploy-Fabricpplication.ps1 scrip in VS.

**Comment by MikkelHegn** [comment]
@alexwun: What's the max timeout of Register-ServiceFabricApplicationType? Is there a max in the runtime that @dmaltsiniotis is hitting?

**Comment by MikkelHegn** [comment]
@dmaltsiniotis in 5.4 the maximum timeout on Register-ServiceFabricApplicationType is 30 min.

Can you confirm that you run with that parameter, but still see this error?
_Start-ServiceFabricApplicationUpgrade : Client timeout must be less than or equal to the cluster configuration for MaxOperationTimeout. Requested value and timeout limit: (29:59.999, 10:00.000)_

**Comment by alexwun** [comment]
The timeout limit in the cluster can be changed via fabric settings for the Service Fabric resource in the ARM template:

"fabricSettings": [
      {
        "name": "NamingService",
        "parameters": [
          {
            "name": "MaxOperationTimeout",
            "value": "<desired max in seconds>"
          },
        ]
      }
],

We will be removing the limit in an upcoming release, at which point you won't have to set that anymore.

**Comment by dmaltsiniotis** [comment]
Hi @MikkelHegn thanks for the quick responses! 30 minutes would be more than sufficient for us, but even passing in 30 minutes I still get an error on Register-ServiceFabricApplicationType. When I was playing around with Visual Studio deploys, I was getting the same error on Start-ServiceFabricApplicationUpgrade.

Here is log from my build:

Publish-UpgradedServiceFabricApplication invoked with timeout values: CopyPackageTimeoutSec: 1800 seconds and RegisterPackageTimeoutSec: 1800 seconds.
Copying application package to image store...
Copy application package succeeded
Registering application type...
Register-ServiceFabricApplicationType : Client timeout must be less than or equal to the cluster configuration for MaxOperationTimeout. Requested value and timeout limit: (29:59.999, 10:00.000)

I'm running 2.4.145 SDK on the server this is hosted on, and the remote Service Fabric cluster is hosted in Azure and is running version 5.4.145.9494.

I apologize if this should have been reported as a new issue, this thread just seemed very close what I was experiencing.

@alexwun Thank you, I will do that now and combine that with these updated scripts and hopefully, we'll have a happy healthy deployment process again!

**Comment by ashoksoo** [comment]
Hi - 

Any update on the compressed packaging ? I have a 300mb package and it takes long time (15 mins) to copy and most of the times it times out. It is a frustrating experience .

**Comment by mholo65** [comment]
@ashoksoo this have been released in Service Fabric 5.5 See blog post [here](https://azure.microsoft.com/en-us/blog/announcing-azure-service-fabric-5-5-and-sdk-2-5/)

just add this to your publish profile
```
<CopyPackageParameters CompressPackage="true" />
```

**Comment by ashoksoo** [comment]
@mholo65 : Thanks for your response. 

Adding it in my cloud.xml publish profile is resulting in an XML error. Can you tell me which section of the publish profile it goes into ? Do you have an example file or something ? I am using Service Fabric 5.5 .

**Comment by MikkelHegn** [comment]
Copy/Pasting from the release notes :-)

Packaging Compression and Copy Timeout
The publish profile schema has been updated to include a <CopyPackageParameters> element which can contain attributes of CompressPackage and CopyPackageTimeoutSec.
<CopyPackageParameters CopyPackageTimeoutSec="3600" CompressPackage="true" />

**Comment by ashoksoo** [comment]
@MikkelHegn : 

I have Service Fabric 5.5 installed and when I add this new tag - it does not recognize it. I dropped and recreated my project to be sure. It does not recognize.

**Comment by MikkelHegn** [comment]
How do you conclude the "is not recognized"? I just validated on my machine with 1.5 tooling extension and 2.5 SDK.

**Comment by ashoksoo** [comment]
I was running on 1.4 tooling. This [blog](https://azure.microsoft.com/en-us/blog/announcing-azure-service-fabric-5-5-and-sdk-2-5/) did not mention about upgrading the tool. It is working once I upgraded the tool by reading this : [paper](https://servicefabricsdkstorage.blob.core.windows.net/public-release-notes/Microsoft%20Azure%20Service%20Fabric%20Release%20Notes%20-%20SDK%202.5.216%20-%20Runtime%205.5.216.docx) . 

Thanks for your help @MikkelHegn  & @mholo65 .

**Comment by oanapl** [comment]
@brutaldev , have you tried compress + [diff provisioning](https://docs.microsoft.com/en-us/azure/service-fabric/service-fabric-application-upgrade-advanced#upgrade-with-a-diff-package)?

**Comment by brutaldev** [comment]
@oanapl , sounded promising, but doesn't solve it.

Now the following warning appears when deploying a build:
`The names of some imported commands from the module 'Create-DiffPackage' include unapproved verbs that might make them less discoverable. To find the commands with unapproved verbs, run the Import-Module command again with the Verbose parameter. For a list of approved verbs, type Get-Verb.`

Things still time out, but for different reasons:
```
...
2018-03-20T09:55:52.5602393Z Trying to create diff package...
...
2018-03-20T10:01:17.2960989Z Copying application to image store...
2018-03-20T10:13:39.0160353Z Upload to Image Store succeeded
2018-03-20T10:13:39.0249934Z Registering application type...
2018-03-20T10:44:52.5066992Z Register application type succeeded
2018-03-20T10:44:52.5375934Z Start upgrading application...
2018-03-20T10:46:53.0832493Z Unregistering application type 'XYZ' and version '1.7.0.20180320.1' failed
2018-03-20T10:46:53.0880672Z -Message Operation timed out.
2018-03-20T10:46:55.6089333Z ##[error]Application type and version is still in use
2018-03-20T10:46:55.6976468Z ##[section]Finishing: Deploy Service Fabric Application
```

Not sure what the deal is with the application type and version still in use, this was a brand new build with some config changes, the current live version is 1.7.0.20180315.2.

Going to try a diff package without compression since that error is very similar if not the same thing that happens when just using compression.

**Comment by brutaldev** [comment]
@oanapl - For a larger application type using compression and diff provisioning it still times out after an hour but this time during registration.
```
2018-03-20T11:52:24.2093073Z Copying application to image store...
2018-03-20T12:25:36.9602334Z Upload to Image Store succeeded
2018-03-20T12:25:36.9677414Z Registering application type...
2018-03-20T13:25:37.9588538Z ##[error]Operation timed out.
2018-03-20T13:25:38.0594259Z ##[section]Finishing: Deploy Service Fabric Application
```

Not sure what's happened but it's become impossible to deploy anything, using VSTS, I have to publish from my machine using VS which is simply not going fly.

**Comment by oanapl** [comment]
Regarding
2018-03-20T10:46:53.0832493Z Unregistering application type 'XYZ' and version '1.7.0.20180320.1' failed
@dbreshears can comment more, but I believe that's a red herring, as we first try to remove existing app type, and this one doesn't exist. We should treat ApplicationTypeNotFound as success.

Re. the timeout during provision, we are changing our scripts to allow users to pass in application type register and unregister timeouts, just like copy package timeout. Meanwhile, you can manually change the scripts to pass the timeouts to the Register-ServiceFabricApplicationType and Unregister-ServiceFabricApplicationType commands.

What kind of VMs are you using in the cluster? We've seen these timeouts when the VMs were slow, either overloaded or not respecting [the minimum requirements](https://docs.microsoft.com/en-us/azure/service-fabric/service-fabric-cluster-capacity#primary-node-type---capacity-guidance). From the article:
"The minimum supported use VM SKU is **Standard D1** or **Standard D1_V2** or equivalent with a minimum of 14 GB of local SSD."

**Comment by brutaldev** [comment]
We are using the minimum **Standard D1_V2** size in the scale set, the CPU usage floats between 70% and 90% currently averaging 83% for the day but never maxing out. The cluster is running 2 applications, 24 services and 57 replicas. At it's peak it runs 3 applications, 40+ services and 100+ replicas.

These are all .NET Core 2.0 stateless micro-services using the latest SF packages with very little in-cluster processing requirements but tons and tons of network usage 24 hours a day.

I'll wait until the register and unregister timeouts become available in the VSTS deployment task, I'm not all too excited to log onto dozens of nodes to makes these updates manually....

I'll scale up the VM sizes and see if that helps and respond. Seems a bit odd to have to have spare resources that remain idle for long periods of time just to facilitate a deployment once in a while. The configuration in place now is absolutely optimal in terms of resource requirements for the services, maximizing our output and minimizing the overall compute cost.

**Comment by bishal-pdMSFT** [comment]
@brutaldev regarding your comment [here](https://github.com/Azure/service-fabric-issues/issues/10#issuecomment-374585331)

1. Looks like you are using a timeout of 1 hours for registering application type. @oanapl VSTS task already supports specifying register timeout. And it seems 1 hour is not enough here.
2. It seems the error message is not conveyed properly. Can you please attach debug logs for this task? Are you using compression and diff?

**Comment by brutaldev** [comment]
The last attempt I tried with the following settings:
- Compress Package: true
- Use Diff Package: true
- CopyPackageTimeoutSec: 3600
- RegisterPackageTimeoutSec: 3600
- Overwrite Behavior: SameAppTypeAndVersion
- Skip upgrade for same type and version: true
- Skip package validation: true (gets performed during the build)

The same thing occurs with all flags set to `false` with long timeouts as well.

Below is the log for the deploy task with private/non-essential/path information omitted, same time out as always these days even though it's a brand new build.

```
2018-03-20T15:49:02.1624650Z ##[section]Starting: Deploy Service Fabric Application
2018-03-20T15:49:02.1631007Z ==============================================================================
2018-03-20T15:49:02.1631433Z Task         : Service Fabric Application Deployment
2018-03-20T15:49:02.1631775Z Description  : Deploy a Service Fabric application to a cluster.
2018-03-20T15:49:02.1632100Z Version      : 1.7.1
2018-03-20T15:49:02.1632406Z Author       : Microsoft Corporation
2018-03-20T15:49:02.1632747Z Help         : [More Information](https://go.microsoft.com/fwlink/?LinkId=820528)
2018-03-20T15:49:02.1633146Z ==============================================================================
2018-03-20T15:49:08.4386091Z Searching for path: D:\a\r1\a\**\projectartifacts\**\PublishProfiles\Cloud.xml
2018-03-20T15:49:08.7031498Z Found path: D:\a\r1\a\xyz - Service Fabric\PRIVATE.servicefabric.xyz\projectartifacts\src\Services\xyz\PublishProfiles\Cloud.xml
2018-03-20T15:49:08.9570518Z Searching for path: D:\a\r1\a\**\applicationpackage
2018-03-20T15:49:09.0749147Z Found path: D:\a\r1\a\xyz - Service Fabric\PRIVATE.servicefabric.xyz\applicationpackage
2018-03-20T15:49:09.2471252Z Imported cluster client certificate with thumbprint '********'.
2018-03-20T15:49:09.2768281Z 
2018-03-20T15:49:09.2772347Z Thumbprint                                Subject                                                                      
2018-03-20T15:49:09.2772957Z ----------                                -------                                                                      
2018-03-20T15:49:09.2773580Z ********  CN=PRIVATE.co.za                                                    
2018-03-20T15:49:17.9056327Z Successfully connected to cluster.
2018-03-20T15:49:17.9378376Z Searching for path: D:\a\r1\a
2018-03-20T15:49:18.0744999Z No items were found with search pattern D:\a\r1\a.
2018-03-20T15:49:18.6908613Z ##[warning]The names of some imported commands from the module 'Create-DiffPackage' include unapproved verbs that might make them less discoverable. To find the commands with unapproved verbs, run the Import-Module command again with the Verbose parameter. For a list of approved verbs, type Get-Verb.
2018-03-20T15:49:18.7312943Z Trying to create diff package...
2018-03-20T15:49:19.8220742Z Copying D:\a\r1\a\xyz - Service Fabric\PRIVATE.servicefabric.xyz\applicationpackage\ApplicationManifest.xml from full application package to C:\Users\VssAdministrator\AppData\Local\Temp\DiffPackage\ApplicationManifest.xml in diff package...
2018-03-20T15:49:20.4714898Z Creating diff package for service: Web.ApiPkg, clusterServiceManifest.Version: 1.1.0.20180315.2, localServiceManifest.Version: 1.1.0.20180320.2
2018-03-20T15:49:20.5116666Z Copying D:\a\r1\a\xyz - Service Fabric\PRIVATE.servicefabric.xyz\applicationpackage\Web.ApiPkg\Code from full application package to C:\Users\VssAdministrator\AppData\Local\Temp\DiffPackage\Web.ApiPkg\Code in diff package...
2018-03-20T15:49:28.1601722Z Copying D:\a\r1\a\xyz - Service Fabric\PRIVATE.servicefabric.xyz\applicationpackage\Web.ApiPkg\ServiceManifest.xml from full application package to C:\Users\VssAdministrator\AppData\Local\Temp\DiffPackage\Web.ApiPkg\ServiceManifest.xml in diff package...
2018-03-20T15:49:28.1795138Z Creating diff package for service: Services.xyz4Pkg, clusterServiceManifest.Version: 1.1.0.20180315.2, localServiceManifest.Version: 1.1.0.20180320.2
2018-03-20T15:49:28.1815985Z Copying D:\a\r1\a\xyz - Service Fabric\PRIVATE.servicefabric.xyz\applicationpackage\Services.xyz4Pkg\Code from full application package to C:\Users\VssAdministrator\AppData\Local\Temp\DiffPackage\Services.xyz4Pkg\Code in diff package...
2018-03-20T15:49:32.1956118Z Copying D:\a\r1\a\xyz - Service Fabric\PRIVATE.servicefabric.xyz\applicationpackage\Services.xyz4Pkg\ServiceManifest.xml from full application package to C:\Users\VssAdministrator\AppData\Local\Temp\DiffPackage\Services.xyz4Pkg\ServiceManifest.xml in diff package...
2018-03-20T15:49:32.2130536Z Creating diff package for service: Services.PersistencePkg, clusterServiceManifest.Version: 1.1.0.20180315.2, localServiceManifest.Version: 1.1.0.20180320.2
2018-03-20T15:49:32.2144508Z Copying D:\a\r1\a\xyz - Service Fabric\PRIVATE.servicefabric.xyz\applicationpackage\Services.PersistencePkg\Code from full application package to C:\Users\VssAdministrator\AppData\Local\Temp\DiffPackage\Services.PersistencePkg\Code in diff package...
2018-03-20T15:49:36.8005672Z Copying D:\a\r1\a\xyz - Service Fabric\PRIVATE.servicefabric.xyz\applicationpackage\Services.PersistencePkg\ServiceManifest.xml from full application package to C:\Users\VssAdministrator\AppData\Local\Temp\DiffPackage\Services.PersistencePkg\ServiceManifest.xml in diff package...
2018-03-20T15:49:37.3178177Z Creating diff package for service: PRIVATE.xyz3Pkg, clusterServiceManifest.Version: 1.1.0.20180315.2, localServiceManifest.Version: 1.1.0.20180320.2
2018-03-20T15:49:37.3179839Z Copying D:\a\r1\a\xyz - Service Fabric\PRIVATE.servicefabric.xyz\applicationpackage\PRIVATE.xyz3Pkg\Code from full application package to C:\Users\VssAdministrator\AppData\Local\Temp\DiffPackage\PRIVATE.xyz3Pkg\Code in diff package...
2018-03-20T15:49:41.7084712Z Copying D:\a\r1\a\xyz - Service Fabric\PRIVATE.servicefabric.xyz\applicationpackage\PRIVATE.xyz3Pkg\ServiceManifest.xml from full application package to C:\Users\VssAdministrator\AppData\Local\Temp\DiffPackage\PRIVATE.xyz3Pkg\ServiceManifest.xml in diff package...
2018-03-20T15:49:41.7121789Z Creating diff package for service: PRIVATE.MakroPkg, clusterServiceManifest.Version: 1.1.0.20180315.2, localServiceManifest.Version: 1.1.0.20180320.2
2018-03-20T15:49:41.7203976Z Copying D:\a\r1\a\xyz - Service Fabric\PRIVATE.servicefabric.xyz\applicationpackage\PRIVATE.MakroPkg\Code from full application package to C:\Users\VssAdministrator\AppData\Local\Temp\DiffPackage\PRIVATE.MakroPkg\Code in diff package...
2018-03-20T15:49:45.6371084Z Copying D:\a\r1\a\xyz - Service Fabric\PRIVATE.servicefabric.xyz\applicationpackage\PRIVATE.MakroPkg\ServiceManifest.xml from full application package to C:\Users\VssAdministrator\AppData\Local\Temp\DiffPackage\PRIVATE.MakroPkg\ServiceManifest.xml in diff package...
2018-03-20T15:49:45.6534387Z Creating diff package for service: PRIVATE.PickNPayPkg, clusterServiceManifest.Version: 1.1.0.20180315.2, localServiceManifest.Version: 1.1.0.20180320.2
2018-03-20T15:49:46.3105587Z Copying D:\a\r1\a\xyz - Service Fabric\PRIVATE.servicefabric.xyz\applicationpackage\PRIVATE.PickNPayPkg\Code from full application package to C:\Users\VssAdministrator\AppData\Local\Temp\DiffPackage\PRIVATE.PickNPayPkg\Code in diff package...
2018-03-20T15:49:48.4720910Z Copying D:\a\r1\a\xyz - Service Fabric\PRIVATE.servicefabric.xyz\applicationpackage\PRIVATE.PickNPayPkg\ServiceManifest.xml from full application package to C:\Users\VssAdministrator\AppData\Local\Temp\DiffPackage\PRIVATE.PickNPayPkg\ServiceManifest.xml in diff package...
2018-03-20T15:49:48.4762307Z Creating diff package for service: PRIVATE.xyz2Pkg, clusterServiceManifest.Version: 1.1.0.20180315.2, localServiceManifest.Version: 1.1.0.20180320.2
2018-03-20T15:49:48.4777862Z Copying D:\a\r1\a\xyz - Service Fabric\PRIVATE.servicefabric.xyz\applicationpackage\PRIVATE.xyz2Pkg\Code from full application package to C:\Users\VssAdministrator\AppData\Local\Temp\DiffPackage\PRIVATE.xyz2Pkg\Code in diff package...
2018-03-20T15:49:52.1711235Z Copying D:\a\r1\a\xyz - Service Fabric\PRIVATE.servicefabric.xyz\applicationpackage\PRIVATE.xyz2Pkg\ServiceManifest.xml from full application package to C:\Users\VssAdministrator\AppData\Local\Temp\DiffPackage\PRIVATE.xyz2Pkg\ServiceManifest.xml in diff package...
2018-03-20T15:49:52.1875962Z Creating diff package for service: PRIVATE.ShopritePkg, clusterServiceManifest.Version: 1.1.0.20180315.2, localServiceManifest.Version: 1.1.0.20180320.2
2018-03-20T15:49:52.1889356Z Copying D:\a\r1\a\xyz - Service Fabric\PRIVATE.servicefabric.xyz\applicationpackage\PRIVATE.ShopritePkg\Code from full application package to C:\Users\VssAdministrator\AppData\Local\Temp\DiffPackage\PRIVATE.ShopritePkg\Code in diff package...
2018-03-20T15:49:56.2258146Z Copying D:\a\r1\a\xyz - Service Fabric\PRIVATE.servicefabric.xyz\applicationpackage\PRIVATE.ShopritePkg\ServiceManifest.xml from full application package to C:\Users\VssAdministrator\AppData\Local\Temp\DiffPackage\PRIVATE.ShopritePkg\ServiceManifest.xml in diff package...
2018-03-20T15:49:56.2467981Z Creating diff package for service: PRIVATE.xyz1Pkg, clusterServiceManifest.Version: 1.1.0.20180315.2, localServiceManifest.Version: 1.1.0.20180320.2
2018-03-20T15:49:56.2481459Z Copying D:\a\r1\a\xyz - Service Fabric\PRIVATE.servicefabric.xyz\applicationpackage\PRIVATE.xyz1Pkg\Code from full application package to C:\Users\VssAdministrator\AppData\Local\Temp\DiffPackage\PRIVATE.xyz1Pkg\Code in diff package...
2018-03-20T15:50:02.1179761Z Copying D:\a\r1\a\xyz - Service Fabric\PRIVATE.servicefabric.xyz\applicationpackage\PRIVATE.xyz1Pkg\ServiceManifest.xml from full application package to C:\Users\VssAdministrator\AppData\Local\Temp\DiffPackage\PRIVATE.xyz1Pkg\ServiceManifest.xml in diff package...
2018-03-20T15:50:03.9737494Z Copying application to image store...
2018-03-20T15:52:53.6425444Z Upload to Image Store succeeded
2018-03-20T15:52:53.6482691Z Registering application type...
2018-03-20T16:03:36.3157356Z Register application type succeeded
2018-03-20T16:03:36.3474942Z Start upgrading application...
2018-03-20T16:05:36.8820126Z Unregistering application type 'xyz' and version '1.7.0.20180320.2' failed
2018-03-20T16:05:36.8871501Z -Message Operation timed out.
2018-03-20T16:05:39.9468720Z ##[error]Application type and version is still in use
2018-03-20T16:05:40.0699389Z ##[section]Finishing: Deploy Service Fabric Application
```

**Comment by oanapl** [comment]
@bishal-pdMSFT , this doesn't seem right:

2018-03-20T15:50:03.9737494Z Copying application to image store...
2018-03-20T15:52:53.6425444Z Upload to Image Store succeeded
-> looks like copy took **2 minutes**.

2018-03-20T15:52:53.6482691Z Registering application type...
2018-03-20T16:03:36.3157356Z Register application type succeeded
-> register took about **10-11 minutes**

2018-03-20T16:03:36.3474942Z Start upgrading application...
-> did upgrade complete? can VSTS print the status for the upgrade operation (periodically print how many UDs are completed/remaining)?

2018-03-20T16:05:36.8820126Z Unregistering application type 'xyz' and version '1.7.0.20180320.2' failed
2018-03-20T16:05:36.8871501Z -Message Operation timed out.
-> this returns immediately, Could it be that the upgrade is still in progress, and the unregister can't be processed? If that's the case, instead of initial message, it prints timeout incorrectly? 

2018-03-20T16:05:39.9468720Z ##[error]Application type and version is still in use
-> this seems like the correct error, app type and version in use.

**Comment by brutaldev** [comment]
@oanapl, The scale up operation put the cluster into a completely broken state. Almost all services are either in an unhealthy state or not running. Services aren't reporting errors but things have largely crawled to a halt and even the web API's are timing out.

The image store service is bust and releases are still timing out as usual. This is the message in the cluster manager for the image store:
`Unhealthy event: SourceId='System.RAP', Property='IReplicator.CloseDuration', HealthState='Warning', ConsiderWarningAsError=false.
The api IReplicator.Close on node _Primary_0 is stuck. Start Time (UTC): 2018-03-20 19:20:17.354.`

>  did upgrade complete?

Nope, I've been unable to upgrade for a few weeks now.
> this seems like the correct error, app type and version in use.

Not really, this version does not exist at all, this was a brand new build to test.

Seems like my only options at this point is delete all applications, restart the VM's to clear system errors and redeploy from scratch. All services are stateless so I won't lose anything but we're going to experience a bit of downtime.

**Comment by oanapl** [comment]
@brutaldev , how did you execute the scale-up? Did you follow the instructions in our documentation?
https://docs.microsoft.com/en-us/azure/service-fabric/service-fabric-cluster-scale-up-down  
Did you add more VMs and then removed the existing ones?
Please note that my initial question was about using the SKU's we don't support. I've seen people trying lower SKUs, which don't guarantee any perf. I never said you should increase the SKU you have since they meet our mim requirements. 
At this point your issues are different than the issue initially reported on this thread. Let's stop overusing this issue, please open another one so we can correctly follow up. This way, people won't get confused reading this thread. 
You can also open a support ticket to get more hands-on help to recover this cluster.

**Comment by brutaldev** [comment]
@oanapl  I simply changed the size on the scale set which I guess is what has destroyed the entire cluster. Nothing has worked for over 12 hours now and I cannot repair the cluster at all. The image store service is still stuck, the applications are in a "deleted" state since this morning, and I cannot deploy either so I'm totally screwed. I requested support on Twitter but haven't heard anything yet. I might just go ahead and delete the whole cluster and recreate it, I have the deployment scripts. This has all been a complete nightmare so in the end I'll probably be looking at alternatives since SF seems overly complicated and falls apart at the slightest hiccup (this is far from the first time we've had disasters with it in the past year).

**Comment by dmaltsiniotis** [comment]
@brutaldev Opening a high severity support ticket/case with Microsoft really is the best way to get support for this issue. I don't think MSFT monitors twitter for support issues. Also, this GitHub issue is being used as pseudo-support forum, which is also probably not appropriate.

In the interest of trying to help though: Have you tried to manually delete/drop all the application replica sets? That may forcefully remove them from their deleted state, but it will leave files and processes potentially left over on each VM. After dropping all of the replica-sets, I would RDP in to each individual VM and reboot it, I have found this to sometimes be an effective brute-force way of "resetting" a cluster back to a working state. Also, slow disk performance on the VM's exacerbates Service Fabric problems DRAMATICALLY. make sure your VMs are not suffering from a bad storage driver configuration: https://support.microsoft.com/en-us/help/4056345/poor-performance-for-vms-that-run-with-an-emulated-mode-storage-stack

**Delete all replica sets PowerShell script:**
`Connect-ServiceFabricCluster -ConnectionEndpoint "yourinstance.region.cloudapp.azure.com:19000" -X509Credential -ServerCommonName "common_name_of_your_ssl_cert" -StoreLocation "CurrentUser" -StoreName "MY" -FindType "FindByThumbprint" -FindValue "SSLTHUMBPRINT" -ServerCertThumbprint "SSLTHUMBPRINT"
$apps = Get-ServiceFabricApplication
$nodes = Get-ServiceFabricNode
foreach($node in $nodes)
{
    foreach($app in $apps)
    {
        $replicas = Get-ServiceFabricDeployedReplica -NodeName $node.NodeName -ApplicationName $app.ApplicationName
        foreach ($replica in $replicas)
        {
            Write-Host $node.NodeName $replica.Partitionid $replica.ReplicaOrInstanceId $replica.ServiceName
            Remove-ServiceFabricReplica -ForceRemove -NodeName $node.NodeName -PartitionId $replica.Partitionid -ReplicaOrInstanceId $replica.ReplicaOrInstanceId
        }
    }
}`

**Comment by brutaldev** [comment]
@dmaltsiniotis Thanks for the info, I'll look at the VM storage driver config once I have things back up.
The script provided had no effect, the applications are still in the "Deleting" state as they have been for over 12 hours now.

`Get-ServiceFabricDeployedReplica` does not return anything for any of the nodes or applications. I tried using `Remove-ServiceFabricApplication` instead and now I'm back to timeouts...

```
Remove-ServiceFabricApplication -ApplicationName fabric:/XYZ-ForceRemove
Remove-ServiceFabricApplication : Operation timed out.
At line:1 char:1
+ Remove-ServiceFabricApplication -ApplicationName fabric:/XYZ -ForceRemove
```

@oanapl  Twitter support provided me a free support ticket to use through the portal. I'm happy to clear out all the VM and just release the latest version in the pipeline onto new VM's. Right now I'm treading carefully as not to make the situation much worse than it already is.

**Comment by bishal-pdMSFT** [comment]
@brutaldev regarding the VSTS issue mentioned [earlier](https://github.com/Azure/service-fabric-issues/issues/10#issuecomment-374736379), I have created an issue in VSTS task repo. Let's carry our conversation there rather than polluting this SF specific thread.

**Comment by brutaldev** [comment]
@bishal-pdMSFT Please give me a link to the issue in the VSTS repo.
@oanapl After a full cluster re-deploy, the first release worked to get the applications/services running again. However, I'm back to timeouts again trying to perform an upgrade so I give up...
```
2018-03-22T08:17:00.4684210Z Copying application to image store...
2018-03-22T08:26:53.4195841Z Upload to Image Store succeeded
2018-03-22T08:26:53.4259941Z Registering application type...
2018-03-22T08:36:13.5563379Z Register application type succeeded
2018-03-22T08:36:13.5902441Z Start upgrading application...
2018-03-22T08:38:14.1842263Z Unregistering application type 'XYZ' and version '1.4.3.20180322.1' failed
2018-03-22T08:38:14.1899343Z -Message Operation timed out.
2018-03-22T08:38:14.7500492Z ##[error]Application type and version is still in use
2018-03-22T08:38:14.8135210Z ##[section]Finishing: Deploy Service Fabric Application
```

Just as before, this version does not exist so I have no idea what it's trying to "unregister".
Not sure if this is related to the VSTS task or the deployment scripts on the cluster.

**Comment by bishal-pdMSFT** [comment]
https://github.com/Microsoft/vsts-tasks/issues/6761

**Comment by oanapl** [comment]
@brutaldev , let's use https://github.com/Microsoft/vsts-tasks/issues/6761 to continue investigation.
@bishal-pdMSFT said that the start upgrade itself fails, and the error printed is incorrect, this is the bug he is tracking in vsts task.

**Comment by brutaldev** [comment]
The timeout settings don't appear to have any effect, unregistering times out after a minutes.
```
2018-03-23T11:15:37.2963642Z Application type 'XYZ' and version '1.4.3.20180323.1' was already registered with the cluster, unregistering it...
2018-03-23T11:17:38.0388046Z ##[error]Operation timed out.
2018-03-23T11:17:38.0946161Z ##[section]Finishing: Deploy Service Fabric Application
```
I give up, we're only using a light integration with Service Fabric, mainly the inter-service communication (which I had to [re-write the JSON implementation docs](https://github.com/MicrosoftDocs/azure-docs/pull/4380)) for because they were so faulty so I'd rather we just implement that ourselves. Going to abandon Service Fabric and just run on a large VM until further notice. Updates are far too slow and with all these deployment issues we've wasted weeks troubleshooting and our clients are getting fed (and so are we).

**Comment by oanapl** [comment]
@brutaldev , again, I am sorry about your bad experience. 

Let me know if you still want to investigate your issue. Were you contacted for the support incident you created? Again, that's the best way to get this resolved. If the cluster is unhealthy, operations won't go through, so we need to fix that first.

Re. your other questions, when you increase the timeouts, you need to respect the conditions we impose. The error message clearly points to what is wrong.
**Overall upgrade timeout must be greater than the combined health check wait and retry timeouts. Requested values (overall timeout, wait, retry): (1:00:00.000, 1:00:00.000, 1:00:00.000)**
We reject your input, so the start upgrade operation fails on the client. Unregistering is not your problem, as I mentioned before, we do that as a cleanup because the previous operation failed.

**Comment by brutaldev** [comment]
I figured that after playing around with the timings a bit, pity this validation isn't in the task interface itself so you only find out about them at runtime (another improvement perhaps for @bishal-pdMSFT?).

Haven't needed to deploy for a while now, things seem stable again for now. I never ended creating the incident, had the problems persisted for longer I would have gone that route. I suspect all the cluster problems compounded the deployment pipeline issues as well.

**Comment by mkosieradzki** [comment]
@brutaldev Do you think this one might be related: #573 ?

**Comment by brutaldev** [comment]
@mkosieradzki That does sound like it could be contributing to the problem as well, nice find. I'll watch the progress of that issue as well.

**Comment by bishal-pdMSFT** [comment]
Regarding [this](https://github.com/Azure/service-fabric-issues/issues/10#issuecomment-376111038), I believe this should ideally be done at platform or SDK layer while accepting Start-ServiceFabricApplicationUpgrade command. @oanapl do you agree?

**Comment by oanapl** [comment]
@bishal-pdMSFT , the link goes back to this issue, so I am not sure what you are referring to. If you are talking about the time parameters for start upgrade, VS could probably help users with at least some info about what the values should be and the relationship between them. Finding out that the values are not correct only after the package was copied and registered is not ideal.

**Comment by bishal-pdMSFT** [comment]
Agree. However, I am worried about such validations done by VSTS layer getting out-dated if platform changes its behavior. Are these timeout related errors documented somewhere which will give me confidence to rely on them. I found [this ](https://docs.microsoft.com/en-us/azure/service-fabric/service-fabric-application-upgrade-troubleshooting#incorrect-time-outs-are-specified) but it is in troubleshoot steps and might be prone to breaking changes.

**Comment by oanapl** [comment]
That's a valid concern. Our policy is to be backward compat, so we most probably won't change the validation for the parameters. 
Re. documentation, I will add this constraint in [Application upgrade parameters](https://docs.microsoft.com/en-us/azure/service-fabric/service-fabric-application-upgrade-parameters)

**Comment by ddinsoft** [comment]
Hi 
I have the same issue, while copying the application package file to Fabric Image Store I am getting bellow error:
Unable to send file create message for operationId:d4134781-fe56-4d8f-9419-c15bfd346fc0 storeRelativePath:AXSFType-1.0.20181102\AXSF\Code\apps\RetailDesignerHost\DevExpress.XtraEditors.v11.2.dll.deploy error:ConnectionConfirmWaitExpired

Resending the file again for operationId:d4134781-fe56-4d8f-9419-c15bfd346fc0 storeRelativePath:AXSFType-1.0.20181102\AXSF\Code\apps\RetailDesignerHost\DevExpress.XtraEditors.v11.2.dll.deploy after a delay of 03.000 seconds, retryAttempt:0


File Create upload session message didn't get any response for operationId:d4134781-fe56-4d8f-9419-c15bfd346fc0 storeRelativePath:AXSFType-1.0.20181102\AXSF\Code\apps\RetailDesignerHost\DevExpress.XtraEditors.v11.2.dll.deploy error:FABRIC_E_TIMEOUT sendAttemptCount:9 WaitTime:60000 atleastOneCreateChunkResponse:true

**Comment by oanapl** [comment]
@ddinsoft , your issue is a bit different. Since this thread has long been closed, please open a new issue to avoid confusing issues.
